crates/optimism/txpool/src/validator.rs

```diff
@@ -196,6 +196,43 @@ where
         self.apply_op_checks(outcome)
     }
 
+    /// Validates a single transaction with a provided state provider.
+    ///
+    /// This allows reusing the same state provider across multiple transaction validations.
+    ///
+    /// See also [`TransactionValidator::validate_transaction`]
+    ///
+    /// This behaves the same as [`EthTransactionValidator::validate_one_with_state`], but in
+    /// addition, ensures that the account has enough balance to cover the L1 gas cost.
+    pub async fn validate_one_with_state(
+        &self,
+        origin: TransactionOrigin,
+        transaction: Tx,
+        state: &mut Option<Box<dyn StateProvider>>,
```

Comment:
```suggestion
        state: Option<&mut Box<dyn StateProvider>>,
```
ah, also let's change the body of `validate_one` to delegate to this method with `None` passed as state param. for this to be idiomatic we need this change.

Reply:
I don't think this is feasible `validate_one_with_provider` takes mutable referance of `Option<Box<dyn StateProvider>>`

https://github.com/paradigmxyz/reth/blob/ceaf968705306859a315f39caece39c1cbecfd88/crates/transaction-pool/src/validate/eth.rs#L185-L195

Reply:
Already done in https://github.com/paradigmxyz/reth/pull/15342/commits/bb1cbb99dd8d13e1e17103660d3ecff4d984722e @emhane.